### PR TITLE
Complex inte test

### PR DESCRIPTION
**Pull request by antoine-de**
test with mock chaos disruption

### CODE REVIEW

File: source/jormungandr/jormungandr/interfaces/v1/fields.py [hunk 1]
```diff
@@ -367,7 +367,7 @@ def output(self, key, obj):
     "application_periods": NonNullList(NonNullNested(period)),
     "status": fields.String(),
     "updated_at": DateTime(),
-    "tags": NonNullList(NonNullNested(fields.String())),
+    "tags": fields.String(),
```
Comment (kinnou02) [inline source/jormungandr/jormungandr/interfaces/v1/fields.py]: a list seem a good thing

File: source/jormungandr/tests/chaos_disruptions_tests.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+        self._connections.add(producer.connection)
+        return producer
+
+    def setup(self):
+        self.mock_chaos_connection = BrokerConnection("pyamqp://guest:guest@localhost:5672")
+        self._connections = set([self.mock_chaos_connection])#set of connection for the heartbeat
+        self._exchange = Exchange('navitia', durable=True, delivry_mode=2, type='topic')
+        self.mock_chaos_connection.connect()
+
+    def teardown(self):
```
Comment (antoine-de) [inline source/jormungandr/tests/chaos_disruptions_tests.py:57]: need to close connection no ?

File: source/jormungandr/tests/chaos_disruptions_tests.py [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+
+    def send_chaos_disruption(self, disruption_name, impacted_obj, impacted_obj_type):
+        item = make_mock_chaos_item(disruption_name, impacted_obj, impacted_obj_type)
+        with self._get_producer() as producer:
+            producer.publish(item, exchange=self._exchange, routing_key='bob', declare=[self._exchange])
+
+@dataset([("main_routing_test", ['--BROKER.rt_topics=bob', 'spawn_maintenance_worker'])])
+class TestChaosDisruptions(ChaosDisruptionsFixture):
+    """
+    Note: it is done as a new fixture, the spawn a new kraken, in order not the get previous disruptions
```
Comment (l-vincent-l) [inline source/jormungandr/tests/chaos_disruptions_tests.py]: it's not english

File: source/jormungandr/tests/chaos_disruptions_tests.py [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+        stops = get_not_null(response, 'stop_areas')
+        assert len(stops) == 1
+        stop = stops[0]
+        #at first no disruption
+        assert 'disruptions' not in stop
+
+        self.send_chaos_disruption("bob_the_disruption", "stopB", "stop_area")
+
+        #we sleep a bit to let kraken reload the data
+        sleep(0.3)
```
Comment (l-vincent-l) [inline source/jormungandr/tests/chaos_disruptions_tests.py]: It'll be safer to call kraken until there's new data

File: source/jormungandr/tests/chaos_disruptions_tests.py [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+@dataset([("main_routing_test", ['--BROKER.rt_topics=bob', 'spawn_maintenance_worker'])])
+class TestChaosDisruptions(ChaosDisruptionsFixture):
+    """
+    Note: it is done as a new fixture, the spawn a new kraken, in order not the get previous disruptions
+    """
+    def test_disruption_on_stop_area_b(self):
+        """
+        when calling the pt object stopB, at first we have no disruptions,
+
+        then we mock a disruption sent from chaos, and we call again the pt object B
```
Comment (l-vincent-l) [inline source/jormungandr/tests/chaos_disruptions_tests.py]: stopB

File: source/jormungandr/tests/chaos_disruptions_tests.py [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+        stop = stops[0]
+
+        disruptions = get_not_null(stop, 'disruptions')
+
+        impacts_by_uri = {d['uri']: d for d in disruptions}
+
+        #at first we got only one disruption on A
+        assert len(disruptions) == 1
+
+        assert 'bob_the_disruption' in impacts_by_uri
```
Comment (l-vincent-l) [inline source/jormungandr/tests/chaos_disruptions_tests.py]: might be clearer to use an any

File: source/jormungandr/tests/chaos_disruptions_tests.py [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+
+
+@dataset([("main_routing_test", ['--BROKER.rt_topics=bob', 'spawn_maintenance_worker'])])
+class TestChaosDisruptions2(ChaosDisruptionsFixture):
+    """
+    Note: it is done as a new fixture, the spawn a new kraken, in order not the get previous disruptions
```
Comment (l-vincent-l) [inline source/jormungandr/tests/chaos_disruptions_tests.py]: same as above

File: source/jormungandr/tests/chaos_disruptions_tests.py [hunk 8]
```diff
@@ -0,0 +1,222 @@
...
+                      if s['to']['embedded_type'] == 'stop_point' and s['to']['id'] == 'stop_point:stopB']
+
+        assert stops_b_to
+
+        for b in stops_b_to:
+            disruptions = get_not_null(b['stop_area'], 'disruptions')
+
+            impacts_by_uri = {d['uri']: d for d in disruptions}
+
+            #at first we got only one disruption on A
```
Comment (l-vincent-l) [inline source/jormungandr/tests/chaos_disruptions_tests.py:176]: B

File: source/jormungandr/tests/chaos_disruptions_tests.py [hunk 9]
```diff
@@ -0,0 +1,222 @@
...
+    feed_message.header.incrementality = gtfs_realtime_pb2.FeedHeader.DIFFERENTIAL
+    feed_message.header.timestamp = 0
+
+    feed_entity = feed_message.entity.add()
+    feed_entity.id = "toto"
+    feed_entity.is_deleted = False
+
+    disruption = feed_entity.Extensions[chaos_pb2.disruption]
+
+    #disruption = chaos_pb2.Disruption()
```
Comment (l-vincent-l) [inline source/jormungandr/tests/chaos_disruptions_tests.py]: You might want to delete it, unless it's a comment

File: source/jormungandr/tests/disruptions_tests.py [hunk 10]
```diff
@@ -103,9 +103,7 @@ def test_disruption_end_point(self):
 
     def test_disruption_with_stop_areas(self):
         """
-        simple disruption call
-
-        without param we don't get the stop areas, so we'll get the impact on the line and on the network
+        when calling the pt object stopA, we should get it's disruptions
```
Comment (l-vincent-l) [inline source/jormungandr/tests/disruptions_tests.py]: its

**Pull request merged**